runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6032,6 +6060,12 @@ done:;
 			case J9DescriptionCpTypeMethodHandle:
 				resolveMethodHandleRef(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE | J9_RESOLVE_FLAG_NO_CLASS_INIT);
 				break;
+			case J9DescriptionCpTypeConstantDynamic:
+				resolveConstantDynamic(_currentThread, ramConstantPool, index, J9_RESOLVE_FLAG_RUNTIME_RESOLVE);
+				if (((J9RAMConstantDynamicRef*)ramCPEntry)->exception == _vm->voidReflectClass->classObject) {
```

Comment:
This needs a comment to describe why it's correct

Reply:
We should do the Void.class check first and let the unresolved case go around the retry loop again.
Saves a call to `resolveConstantDynamic` on every condy of null